apps/examples/testcase/le_tc/kernel/tc_memory_safety.c

```diff
@@ -0,0 +1,408 @@
+
+	/* Start the receiving thread with priority2 */
+
+	status = pthread_attr_init(&attr);
+	TC_ASSERT_EQ("pthread_attr_init", status, OK);
+
+	status = pthread_attr_setstacksize(&attr, STACKSIZE);
+	TC_ASSERT_EQ("pthread_attr_setstacksize", status, OK);
+
+	sparam.sched_priority = priority2;
+	status = pthread_attr_setschedparam(&attr, &sparam);
+	TC_ASSERT_EQ("pthread_attr_setschedparam", status, OK);
+
+	status = pthread_create(&receiver, &attr, receiver_thread, NULL);
+	TC_ASSERT_EQ("pthread_create", status, OK);
+
+	/* Start the sending thread with priority3 */
+
+	status = pthread_attr_init(&attr);
+	TC_ASSERT_EQ("pthread_attr_init", status, OK);
+
+	status = pthread_attr_setstacksize(&attr, STACKSIZE);
+	TC_ASSERT_EQ("pthread_attr_setstacksize", status, OK);
+
+	sparam.sched_priority = priority3;
+	status = pthread_attr_setschedparam(&attr, &sparam);
+	TC_ASSERT_EQ("pthread_attr_setschedparam", status, OK);
+
+	status = pthread_create(&sender, &attr, sender_thread, NULL);
+	TC_ASSERT_EQ("pthread_create", status, OK);
+
+	/* The test is done. Let's wrap up the test */
+
+	pthread_join(sender, &result);
+	TC_ASSERT_EQ("pthread_join", result, (void *)0);
+
+	g_exit = 1;
+	
+#ifndef CONFIG_DISABLE_SIGNALS
+	/* Wake up the receiver thread with a signal */
+
+	pthread_kill(receiver, 9);
+
+	/* Wait a bit to see if the thread exits on its own */
+
+	usleep(HALF_SECOND_USEC_USEC);
+#endif
+
+	/* Then cancel the thread and see if it did */
+
+	expected = PTHREAD_CANCELED;
+	status = pthread_cancel(receiver);
+	if (status == ESRCH) {
+		tckndbg("receiver has already terminated\n");
+		expected = (FAR void *)0;
+	}
+
+	/* Check the result.  If the pthread was canceled, PTHREAD_CANCELED is the
+	 * correct result.  Zero might be returned if the thread ran to completion
+	 * before it was canceled.
+	 */
+
+	pthread_join(receiver, &result);
+	TC_ASSERT_EQ("pthread_join", result, expected);
+
+	pthread_join(looper, &result);
+	TC_ASSERT_EQ("pthread_join", result, 0);
+
+	/* Message queues are global resources and persist for the life the the
+	 * task group.  The message queue opened by the sender_thread must be closed
+	 * since the sender pthread may have been canceled and may have left the
+	 * message queue open.
+	 */
+
+	if (result == PTHREAD_CANCELED && g_recv_mqfd) {
+		TC_ASSERT_GEQ("mq_close", mq_close(g_recv_mqfd), 0);
+	} else if (result != PTHREAD_CANCELED && g_recv_mqfd) {
+		TC_ASSERT_EQ("mq_close", mq_close(g_recv_mqfd), 0);
+	}
+
+	/* Make sure that the receive queue is closed as well */
+	TC_ASSERT_EQ("mq_close", g_send_mqfd, NULL);
+
+	/* Destroy the message queue */
+	TC_ASSERT_GEQ("mq_unlink", mq_unlink("mqueue"), 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+
+/****************************************************************************
+ * Name: memory_safety
+ ****************************************************************************/
+int memory_safety_main(void)
+{
+	tc_memory_sefety_with_mqueue();
```

Comment:
same as above(should change sefety to safety.)

Reply:
fixed